plugins/inventory-backend/src/service/router.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import express from 'express';
+import { Logger } from 'winston';
+import { AggregatorInventory, StaticInventory } from '../inventory';
+
+export interface RouterOptions {
+  logger: Logger;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const logger = options.logger.child({ plugin: 'inventory' });
+
+  const inventory = new AggregatorInventory();
+  inventory.enlist(
+    new StaticInventory([
+      { id: 'component1' },
+      { id: 'component2' },
+      { id: 'component3' },
+      { id: 'component4' },
+    ]),
+  );
+
+  const router = express.Router();
+  router
+    .get('/', async (req, res) => {
+      const components = await inventory.list();
+      res.status(200).send(components);
+    })
+    .get('/:id', async (req, res) => {
+      const { id } = req.params;
+      const component = await inventory.item(id);
```

Comment:
alternatively, have `intentory.item()` throw a `NotFoundError` which your error handler interprets as 404 and properly sends some standard thing for

Reply:
that could become a useful standard across all the backends tbh

Reply:
We discussed that recently actually.

https://github.com/spotify/backstage/pull/676#discussion_r417516624

I initially copied over the custom resultCode carrying errors from lighthouse, but in the end removed them again. For two reasons:

Firstly, the http-errors library seemed excellent and established, for cases where that fits. Good find by @ Rugvip 

Secondly, I pondered for a bit and wasn't sure we wanted to promote that as a pattern. In general I frown at throwing  HTTP errors from deep business logic code that has nothing to do with HTTP.

Of course we could make a separate business ComponentNotFound error that isn't meant in the HTTP sense. But then I would like to have it caught at the end of this plugin's router chain, instead of letting it leak out to the root of the express app.

And perhaps this isn't an error state at all, but rather one of the normal return values from 'item' - that's the camp I'm in. If this were like Haskell or something, it would simply be encoded in the return type.

But still pondering, and would love to hear other perspectives :)

Reply:
I usually end up having a couple of standard errors in JS, like NotFoundError and ConflictError, end then checking for them by name. So +1 to @fastfrwrd

Reply:
OK, maybe add in a later PR?

Reply:
(and if we do that - does it even make sense to use http-errors despite it being leaky to do so from deep code? like, some database layer throwing `createError.Conflict` imported from http-errors)